fe/fe-core/src/main/java/com/starrocks/sql/analyzer/MaterializedViewAnalyzer.java

```diff
@@ -181,6 +196,51 @@ public Void visitCreateMaterializedViewStatement(CreateMaterializedViewStatement
             return null;
         }
 
+        // TODO(murphy) implement
+        // Plan the query statement and store in memory
+        private void planMVQuery(CreateMaterializedViewStatement createStmt, QueryStatement query, ConnectContext ctx) {
+            if (!ctx.getSessionVariable().isEnableRealtimeRefreshMV()) {
+                return;
+            }
+            if (!createStmt.getRefreshSchemeDesc().getType().equals(MaterializedView.RefreshType.REALTIME)) {
+                return;
+            }
+            QueryRelation queryRelation = query.getQueryRelation();
+
+            ColumnRefFactory columnRefFactory = new ColumnRefFactory();
+            LogicalPlan logicalPlan = new RelationTransformer(columnRefFactory, ctx).transform(queryRelation);
+            Map<ColumnRefOperator, ScalarOperator> columnRefMap = new HashMap<>();
+            List<ColumnRefOperator> outputColumns = new ArrayList<>();
+            for (int colIdx = 0; colIdx < logicalPlan.getOutputColumn().size(); colIdx++) {
+                ColumnRefOperator ref = logicalPlan.getOutputColumn().get(colIdx);
+                outputColumns.add(ref);
+                columnRefMap.put(ref, ref);
+            }
+
+            // Build logical plan for view query
+            OptExprBuilder optExprBuilder = logicalPlan.getRootBuilder();
+            logicalPlan = new LogicalPlan(optExprBuilder, outputColumns, logicalPlan.getCorrelation());
+            Optimizer optimizer = new Optimizer();
+            PhysicalPropertySet requiredPropertySet = PhysicalPropertySet.EMPTY;
+            ExecPlan execPlan;
+            try {
+                ctx.getSessionVariable().setMVPlanner(true);
+                OptExpression optimizedPlan = optimizer.optimize(
```

Comment:
It seems that you need logical plan here? I have done this in query rewrite. I optimize it by using part of optimizer rules

Reply:
I need a physical plan and physical operator. But without apply some rules, such as mv rewrite.

These detailed would be refined later.